#277 racing timer offset and string formatting

Mi-Sad opened the pull request:
Solved the code smell popped up in sonarcloud on the prev pull request

Mi-Sad committed:
Added string formatting to room timers and actually storing the last local result and display it

Added string formatting to room timer so that no weird .0003 would pop up in the timer (racing) and also added some component to actually calcultate the last local result and store it in the var inside racinggamemode, i honeslty don't like this solution but works perfectly for the moment (as what i like would require a total rework of the FGMMKII class which is in a hold state till some other issue are solved)

Mi-Sad committed:
Just solved the code smells popped out in the prev pull request

removed commented code and also added some const val instead of popping up numbers in code

Mi-Sad committed:
Update RacingGamemode.cs

Lithrun commented on Assets/Scripts/FengGameManagerMKII.cs:
```diff
@@ -626,16 +626,10 @@ private void core2()
-                this.ShowHUDInfoTopCenter(content);
-                content = Gamemode.GetGamemodeStatusTopRight((int) timeTotalServer, time);
-                this.ShowHUDInfoTopRight(content);
+                //All of this has to go as it's a double from also the previous if
```
So if you already discovered that this if statement does not have any impact, then please remove it.

Mi-Sad replied:
Ok i'll start reworking those than ;)

Lithrun commented on Assets/Scripts/Gamemode/GamemodeBase.cs:
```diff
@@ -204,6 +204,7 @@ public virtual void OnTitanKilled(string titanName)
             }
         }
 
+        //would be better to get float value for time and totalRoomTime instead as those are float value to begin with so this method won't have any loss of infos
```
Time should be managed via TimeService, but why place a comment here?

Mi-Sad replied:
I'll check this timeservice thingh

Lithrun replied:
TimeService is currently only available in Issue 160 however: https://github.com/AoTTG-2/AoTTG-2/blob/%23160-titan-difficulty/Assets/Scripts/Services/TimeService.cs

Will try to do the partial merge asap as discussed so you can use that one.

Lithrun commented on Assets/Scripts/Gamemode/RacingGamemode.cs:
```diff
@@ -8,12 +8,27 @@ namespace Assets.Scripts.Gamemode
+        /// <summary>
+        /// issue #277 will have to be removed once the conversion from fenggamemanager to single manager will be done
+        /// </summary>
+        public const int IntRoundTimeStatusTopScale = 10;
```
Use [Obsolete] instead of a summary if you intention is that this should not be used in the future

Mi-Sad replied:
yeah, just didn't want to put new stuff directly as obsolete :'D

Mi-Sad replied:
Also cos it wouldn't be atm as the replacement code is not there yet

Lithrun commented on Assets/Scripts/Gamemode/RacingGamemode.cs:
```diff
@@ -22,10 +37,8 @@ public override void OnGameWon()
-                //this.chatRoom.addLINE("<color=#FFC000>(" + this.roundTime.ToString("F2") + ")</color> Round ended (game win).");
-            }
+
+            //there was an if on (int) FengGameManagerMKII.settings[0xf4]) == 1 and just a comment inside
```
Useless code, its good that you removed that, but remove your comment as well then as without that code it makes no sense

Mi-Sad replied:
right

Lithrun commented on Assets/Scripts/Gamemode/RacingGamemode.cs:
```diff
@@ -44,12 +57,36 @@ public override string GetVictoryMessage(float timeUntilRestart, float totalServ
+        {
+            time -= RacingGamemode.IntRoundTimeStatusTopScale*RacingGamemode.StartTimerCountdown;
+            if (time > 0)
+                //if the starting time has passed it return 
```
Your if statement already says that if time is larger than 0, so useless comments (also for the else statement)

Mi-Sad replied:
well just to clarify as time gets an operation done before

Lithrun commented on Assets/Scripts/FengGameManagerMKII.cs:
```diff
@@ -159,6 +159,7 @@ public class FengGameManagerMKII : Photon.MonoBehaviour
     public int time = 600;
     private float timeElapse;
+    //should be better to be able to retrive this info somehow in order to be used in other place
     private float timeTotalServer;
```
Rather mark this with an [Obsolete] attribute instead of a comment. Via obsolete your IDE will display a message when you are trying use the field, for a comment you'd have to look it up here.

Mi-Sad replied:
roger

Lithrun reviewed (changes requested):
- There are comments which do not add much of a value. Instead remove those or use the [Obsolete] attribute (with a description)

Sonarcloud will complain about that there are new [Obsolete] fields / methods used, but you can ignore those as that's something for a refactor issue.

Mi-Sad committed:
Changed the currentcamera type to match the needings

That local var currentCamera is only of local use (changed from public to private) and the type as is only required for IN_GAME_MAIN_CAMERA method, getting rid of a getcomponent each time.

Mi-Sad committed:
Change var type from GameObject to IN_GAME_MAIN_CAMERA

as in playertitan, this local var is only used for that component so using the component directly would get rid of a lot of getcomponent call.

Mi-Sad committed:
Change type GameObject to IN_GAME_MAIN_CAMERA

same as the other 2, is only used for his component IN_GAME_MAIN_CAMERA

Mi-Sad committed:
Moving showHUDInfo from fenggamemanager to the dedicated HUD class

Removing obsolete code before totally breaking down and moving core2 from fenggamemanager to the relative gamemode class.

Mi-Sad committed:
Moved the management of the HUD server and gamemode related from FengGameManager to Relative GameMode class

There are also few tiny changes, like moving some component from Fenggamemanager to the gamemode manager and some few adjustment of the methods of the gamemodeclass to be more suited to operate with the timeservice class

Mi-Sad committed:
Even more breaking a part of the coreupdate in fgm class

Mi-Sad committed:
removed some code duplicate of Hero death's events

Mi-Sad committed:
Added 2 static method just to make writing faster and call RPC on FGM without going through instance and photonview

Mi-Sad committed:
Just a rename cos it was s instead of S

Mi-Sad committed:
Implemented the respawn for racing gamemode as a triggered event instead of as a continuous check

Mi-Sad committed:
Almost completed the migration of racinggamemode component to racinggamemode class

Also replaced the ICompared with Linq as requested by the obsolete tag of the class

Mi-Sad committed:
Moving the creation of the string holding the race result in the trigger event instead of running a check

Mi-Sad committed:
Moving methods from fenggamemanager to gamemode classes

Mi-Sad committed:
Moved some more component to gamemode classes from fgm

Mi-Sad committed:
Update RacingGamemode.cs

Mi-Sad committed:
Merge branch 'development' into #277-Racing-Timer-Offset-And-String-Formatting

Mi-Sad committed:
Merge branch 'development' into #277-Racing-Timer-Offset-And-String-Formatting

Mi-Sad committed:
Merging with current dev

Mi-Sad commented:
Impossible to merge with how the current system is now working, is a shame i had this ready for the merge months ago and waiting just to become obsolete and totally need to be redone from 0, i won't work on this thing twice...

Lithrun commented:
That's a shame indeed. Perhaps for future references, try to finish a PR when you can or allow someone else to takeover, as this one has been a draft for months.
